### PR TITLE
Subscription settings: Turn off the emails check

### LINKED ISSUE #74471: `/subscriptions/settings`: Create “Block emails” input
## Design
inDLaEQV8jJ21O4WXawIsJ-fi-712%3A29263

### Mobile Design
block-emails-input-settings-mobile

### Desktop Design
block-email-input-settings-screen-desktop

## Acceptance Criteria
- display the label with the input as per the Figma design
- the label and the description should both be translated
- have the correct value displayed based on the relevant prop passed into the `UserSettings` container
- Note: There is no need to deal with the submission of the changed value as part of this task. The submission will be implemented with the following task, once all the settings form inputs are being included in the parent component: https://github.com/Automattic/wp-calypso/issues/74433

### PR DESCRIPTION
**Pull request by JuanLucha**
## Proposed Changes

This PR adds the checkbox to turn off the reception of emails to which the user is subscribed. This is the result:

Screenshot 2023-03-15 at 19 38 24

## Testing Instructions

1. Apply this PR and start the application (`yarn start`).
2. Go to `http://calypso.localhost:3000/subscriptions/settings`.
3. You should see the Block Emails setting there:
Screenshot 2023-03-16 at 12 03 35

So that you know, there is no interaction in the option at the moment. When you click on the checkbox, nothing happens. That interactivity will be implemented in other PR.

Thanks for the review! ❤️

Fixes https://github.com/Automattic/wp-calypso/issues/74471

### CODE REVIEW

File: client/my-sites/subscriptions/turn-off-emails.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { useTranslate } from 'i18n-calypso';
```
Comment (mashikag) [inline client/my-sites/subscriptions/turn-off-emails.tsx]: I am sorry I did not make this clear in the issue description, but the idea is that we define all the components dedicated for our SubscriptionManager in the `@automattic/subscription-manager` package. The package is located at [`packages/subscription-manager`](https://github.com/Automattic/wp-calypso/tree/trunk/packages/subscription-manager).

I would suggest naming all the setting components with "Setting" postfix in their name (e.g.: `BlockEmailsSetting`, `EmailFormatSetting` etc.), and to define them in `packages/subscription-manager/src/components/UserSettings`.

File: client/my-sites/subscriptions/turn-off-emails.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import './style.scss';
+
+const TurnOffEmails = ( { areEmailsBlocked } ) => {
+	const translate = useTranslate();
+
+	return (
+		<FormFieldset className="turn-off-emails">
+			<span className="title">Turn off emails</span>
```
Comment (mashikag) [inline client/my-sites/subscriptions/turn-off-emails.tsx]: The title should be "Block emails". inDLaEQV8jJ21O4WXawIsJ-fi-712%3A29014

File: client/my-sites/subscriptions/turn-off-emails.tsx (hunk 1 above)
Comment (JuanLucha) [reply client/my-sites/subscriptions/turn-off-emails.tsx]: No, no, it was my mistake. I saw your previous PR, and I saw the package there. I just ignored it 🤣

File: packages/subscription-manager/src/components/index.d.ts [hunk 3]
```diff
@@ -89,3 +89,23 @@ declare module 'calypso/components/route' {
...
+	export default FormFieldset;
+}
+
+declare module 'calypso/components/forms/form-label' {
+	interface Props {
+		optional?: boolean;
+		required?: boolean;
+	}
+	type LabelProps = LabelHTMLAttributes< HTMLLabelElement >;
+	const FormLabel: FunctionComponent< Props & LabelProps >;
```
Comment (mashikag) [inline packages/subscription-manager/src/components/index.d.ts]: Small one: Everywhere else in the file, we use React.FC

File: packages/subscription-manager/src/components/BlockEmailsSetting/style.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	display: flex;
+	flex-direction: column;
+
+	&.form-fieldset {
+		margin-left: 2px;
+	}
+
+	.title {
+		margin-bottom: 7px;
+		color: #50575e;
```
Comment (mashikag) [inline packages/subscription-manager/src/components/BlockEmailsSetting/style.scss]: We should stick to using the studio constants where possible: `$studio-gray-60` like [here](https://github.com/Automattic/wp-calypso/blob/c613d29ce84c6fbd38221884af1b9000a01de5c1/packages/subscription-manager/src/components/TabsSwitcher/styles.scss#L35).

File: packages/subscription-manager/src/components/UserSettings/UserSettings.tsx [hunk 5]
```diff
@@ -13,7 +15,9 @@ type UserSettingsProps = {
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars -- until we start using any of these props
 const UserSettings = ( { value = {}, loading = false }: UserSettingsProps ) => (
-	<div className="user-settings">User settings will go here</div>
+	<div className="user-settings">
+		<BlockEmailsSetting onChange={ () => null } value={ false } />
```
Comment (mashikag) [inline packages/subscription-manager/src/components/UserSettings/UserSettings.tsx]: - The value prop can already be set with `value.blocked`.
- The `onChange` prop can already be set with `( value ) => ({ blocked: value })` (the way value is obtained will depend on the implementation of the `FormInputCheckbox`).

File: packages/subscription-manager/src/components/BlockEmailsSetting/block-emails-setting.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+const BlockEmailsSetting = ( { value, onChange }: BlockEmailsSettingProps ) => {
+	const translate = useTranslate();
+
+	return (
+		<FormFieldset className="block-emails-setting">
+			<span className="title">Block emails</span>
+			<FormLabel>
+				<FormInputCheckbox name="block_emails_setting" checked={ value } onChange={ onChange } />
```
Comment (mashikag) [inline packages/subscription-manager/src/components/BlockEmailsSetting/block-emails-setting.tsx]: Not sure if we will end up using the name of the input, but to match our API the input should have a name of "blocked"

File: packages/subscription-manager/src/components/BlockEmailsSetting/style.scss [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+	}
+
+	.form-checkbox {
+		box-sizing: border-box;
+		width: 20px;
+		height: 20px;
+		background: #fff;
+		border: 1px solid #a7aaad;
```
Comment (mashikag) [inline packages/subscription-manager/src/components/BlockEmailsSetting/style.scss]: The color is `$studio-gray-20`

File: packages/subscription-manager/src/components/BlockEmailsSetting/block-emails-setting.tsx (hunk 6 above)
Comment (JuanLucha) [reply packages/subscription-manager/src/components/BlockEmailsSetting/block-emails-setting.tsx]: I'll change it 👍

File: packages/subscription-manager/src/components/BlockEmailsSetting/style.scss (hunk 4 above)
Comment (JuanLucha) [reply packages/subscription-manager/src/components/BlockEmailsSetting/style.scss]: Cool. I'll change it 👍

File: packages/subscription-manager/src/components/BlockEmailsSetting/style.scss [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+	.title {
+		margin-bottom: 7px;
+		color: #50575e;
+	}
+
+	.form-checkbox {
+		box-sizing: border-box;
+		width: 20px;
+		height: 20px;
+		background: #fff;
```
Comment (mashikag) [inline packages/subscription-manager/src/components/BlockEmailsSetting/style.scss]: We should use color constant here too

File: packages/subscription-manager/src/components/UserSettings/UserSettings.tsx (hunk 5 above)
Comment (JuanLucha) [reply packages/subscription-manager/src/components/UserSettings/UserSettings.tsx]: Cool, will change it 👍

File: packages/subscription-manager/src/components/index.d.ts (hunk 3 above)
Comment (JuanLucha) [reply packages/subscription-manager/src/components/index.d.ts]: Nice. Will change 👍

File: packages/subscription-manager/src/components/UserSettings/UserSettings.tsx [hunk 9]
```diff
@@ -13,7 +15,9 @@ type UserSettingsProps = {
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars -- until we start using any of these props
 const UserSettings = ( { value = {}, loading = false }: UserSettingsProps ) => (
-	<div className="user-settings">User settings will go here</div>
+	<div className="user-settings">
+		<BlockEmailsSetting onChange={ ( value ) => ( { blocked: value } ) } value={ value.blocked } />
```
Comment (mashikag) [inline packages/subscription-manager/src/components/UserSettings/UserSettings.tsx]: Ops, sorry my suggestion was incorrect. 🙈 It was meant to be:
```jsx
onChange={ ( value ) => onChange?.( { blocked: value }  ) }
```
But first you need to destructure the props for `onChange`.

**Comment by mashikag** [comment]
The "Block emails" title should be black. 
Screen Shot 2023-03-16 at 17 42 56

File: packages/subscription-manager/src/components/index.d.ts [hunk 10]
```diff
@@ -89,3 +89,23 @@ declare module 'calypso/components/route' {
...
+	const FormFieldset: React.FC< InputHTMLAttributes< HTMLInputElement > >;
+	export default FormFieldset;
+}
+
+declare module 'calypso/components/forms/form-label' {
+	interface Props {
+		optional?: boolean;
+		required?: boolean;
+	}
+	type LabelProps = React.FC< LabelHTMLAttributes< HTMLLabelElement > >;
```
Comment (mashikag) [inline packages/subscription-manager/src/components/index.d.ts]: I am sorry I did not spot this the first time around but it looks like here you are trying to define a type for `LabelProps`. That is props, as the name suggests, and not a component. Therefore the type definition should be as follows:
```typescript
type LabelProps = LabelHTMLAttributes< HTMLLabelElement >;
```

Comment (mashikag) [inline packages/subscription-manager/src/components/index.d.ts]:
```suggestion
	type LabelProps = LabelHTMLAttributes< HTMLLabelElement >;
```

File: packages/subscription-manager/src/components/UserSettings/UserSettings.tsx [hunk 11]
```diff
@@ -13,7 +15,9 @@ type UserSettingsProps = {
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars -- until we start using any of these props
 const UserSettings = ( { value = {}, loading = false }: UserSettingsProps ) => (
```
Comment (mashikag) [inline packages/subscription-manager/src/components/UserSettings/UserSettings.tsx]:
```suggestion
const UserSettings = ( { value = {}, loading = false, onChange }: UserSettingsProps ) => (
```

File: packages/subscription-manager/src/components/UserSettings/UserSettings.tsx (hunk 9 above)
Comment (mashikag) [inline packages/subscription-manager/src/components/UserSettings/UserSettings.tsx]:
```suggestion
		<BlockEmailsSetting onChange={ ( value ) => onChange?.( { blocked: value }  ) } value={ value.blocked } />
```

File: packages/subscription-manager/src/components/BlockEmailsSetting/block-emails-setting.tsx [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+ */
+import { useTranslate } from 'i18n-calypso';
+import { ChangeEventHandler } from 'react';
+import FormInputCheckbox from 'calypso/components/forms/form-checkbox';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import './style.scss';
+
+export type BlockEmailsSettingProps = {
+	value: boolean | undefined;
```
Comment (mashikag) [inline packages/subscription-manager/src/components/BlockEmailsSetting/block-emails-setting.tsx]:
```suggestion
	value?: boolean;
```

File: packages/subscription-manager/src/components/fields/BlockEmailsSetting/block-emails-setting.tsx [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+	value?: boolean;
+	onChange: ChangeEventHandler< HTMLInputElement >;
+};
+
+const BlockEmailsSetting = ( { value, onChange }: BlockEmailsSettingProps ) => {
+	const translate = useTranslate();
+
+	return (
+		<FormFieldset className="block-emails-setting">
+			<span className="title">Block emails</span>
```
Comment (mashikag) [inline packages/subscription-manager/src/components/fields/BlockEmailsSetting/block-emails-setting.tsx]:
```suggestion
			<span className="title">{ translate( 'Block emails' ) }</span>
```

**Review by mashikag — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7874263

Thank you @JuanLucha for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7874263).

**Pull request merged**